### ISSUE TITLE
Progressbar type property updates conflicts with ngAnimate causing no animation

### ISSUE THREAD

**Issue by rgaskill** [angular-ui/bootstrap#2670]
When ngAnimate is imported updating a progress bar value and type at the same time results in the progress bar not animating.  Here is an example

http://plnkr.co/edit/uZr9XqsUgyT4sjDTKhLS?p=preview

It seems the best way to fix this is by this by removing ngCass from the bar templates and just update the bar element class directly in the directive.  I will create the pull request if this seems like a reasonable approach.

**Comment by rgaskill** [comment]
@chrisirhc what do I need to do to get this out of purgatory?  :smiley:   This is still an issue with 0.12.0

**Comment by chrisirhc** [comment]
It looks like this was a bug in AngularJS 1.2 . This is now fixed if you upgrade to AngularJS 1.3 . Since our next release (0.13) moves support to AngularJS 1.3 and above, this issue won't be valid once the 0.13 is released.

Reference Plunker with AngularJS 1.3: http://plnkr.co/edit/VNXiMsLlmqAhOwgy7Agt?p=preview

For AngularJS 1.2 users, an alternative fix is to insert the following style in your stylesheet:
```css
.progress-bar.ng-animate {
  -webkit-transition: 0;
  transition: 0;
}
```
Reference Plunker with CSS fix: http://plnkr.co/edit/SAb7P6jcDtcvALL9Ll00?p=preview

**Comment by JuustoMestari** [comment]
This issue seems to be back with the newer Angular versions ( tried 1.4.3 & 1.4.8)
See [Plunker](http://plnkr.co/edit/hFN1Xwy8Cfsgaf3IE5P2?p=preview)
